pkg/sql/vars.go

```diff
@@ -831,6 +831,9 @@ var varGen = map[string]sessionVar{
 	// See https://www.postgresql.org/docs/10/static/runtime-config-client.html#GUC-LOC-TIMEOUT
 	`lock_timeout`: makeCompatIntVar(`lock_timeout`, 0),
 
+	// See https://www.postgresql.org/docs/13/runtime-config-compatible.html
+	`backslash_quote`: makeCompatStringVar(`backslash_quote`, `safe_encoding`, `on`, `off`, `safe_encoding`),
```

Comment:
for this PR, I think we should only allow the values that we actually support.

I think it would be misleading for users if we allow them to set this to a value, but we don't actually implement the behavior described in the postgres docs.

Reply:
so to be a little more specific, I think this line should be:
```
makeCompatStringVar(`backslash_quote`, `safe_encoding`),
```

It also needs a comment saying that if we allow other values for `client_encoding` that are not UTF-8, then we need to update this to either implement `safe_encoding` fully, or change the default value to `on`.


To fully implement `backslash_quote=off`, we would need to update the code here so that it uses this session variable to determine the value of `allowEscape`: https://github.com/cockroachdb/cockroach/blob/36ed61e139271756805545ae5939c40fb32e7e45/pkg/sql/parser/scan.go#L155-L163

But for this PR, I don't think we should fully implement the setting. @RichardJCai @ZhouXing19 -- what do you think?

Reply:
Yeah I don't think we should fully implement it, let's not support `off` for now.

Reply:
@rafiss @RichardJCai Cool, have changed the `extraAllowed` to an empty set, and modified the tests at `logic_test\set`.